ts/features/bonus/cgn/screens/CgnDetailScreen.tsx

```diff
@@ -180,124 +143,85 @@ const CgnDetailScreen = (props: Props): React.ReactElement => {
   };
 
   return (
-    <LoadingSpinnerOverlay
-      isLoading={
-        props.isCgnInfoLoading ||
-        (props.cgnDetails && cardLoading) ||
-        isLoading(props.unsubscriptionStatus)
+    <BonusCardScreenComponent
+      logoUris={logoUris}
+      name={I18n.t("bonus.cgn.name")}
+      organizationName={I18n.t("bonus.cgn.departmentName")}
+      counters={[]}
+      cardBackground={<AnimatedBackground />}
+      status={
+        props.cgnDetails ? <CgnCardStatus card={props.cgnDetails} /> : undefined
+      }
+      cardFooter={
+        <H2
+          style={{
+            textAlign: "center",
+            marginHorizontal: 16,
+            marginBottom: 16
+          }}
+        >
+          {pot.isSome(currentProfile)
+            ? `${currentProfile.value.name} ${currentProfile.value.family_name}`
+            : ""}
+        </H2>
+      }
+      footerCtaPrimary={
+        showDiscoverCta
+          ? {
+              label: getPrimaryactionLabel(canDisplayEycaDetails),
+              onPress: onPressShowCgnDiscounts
+            }
+          : undefined
+      }
+      footerCtaSecondary={
+        showDiscoverCta && canDisplayEycaDetails
+          ? {
+              label: I18n.t("bonus.cgn.detail.cta.eyca.showEycaDiscounts"),
+              accessibilityLabel: I18n.t(
+                "bonus.cgn.detail.cta.eyca.showEycaDiscounts"
+              ),
+              onPress: () =>
+                openWebUrl(EYCA_WEBSITE_DISCOUNTS_PAGE_URL, () =>
+                  IOToast.error(I18n.t("bonus.cgn.generic.linkError"))
+                )
+            }
+          : undefined
```

Comment:
What do you think about extracting this property assignment `footerCtaSecondary` outside of the render component? Just to make it more readable.

Reply:
I personally write callbacks and conditionals inline because it saves the reader from jumping around the file and the writer of finding meaningful names.
Also in this case moving those parameters elsewhere would imply importing more types. In this case it is only a label and function call, they do not deserve code indirection IMHO.

Reply:
Thank you for your perspective. I understand the desire to keep the code concise and avoid unnecessary indirection. However, I believe that extracting the footerCtaSecondary assignment outside of the render method could still be beneficial for several reasons.

First, separating complex property assignments, especially when they involve conditionals, can enhance the readability of the render method. It allows the main render function to focus on the structure of the component, making it easier for others to follow the logic at a glance.

Second, by extracting this logic, we can better isolate and test it if necessary. It also allows us to give meaningful names to these properties, which can improve code clarity and maintainability in the long term.

Lastly, while the concern about importing additional types is valid, in many cases, the benefits of a more organized and readable codebase outweigh the minor overhead of a few extra imports. In fact, this approach can make it easier to manage types and ensure that the code remains scalable and adaptable to future changes.

I believe that these advantages can contribute positively to the overall quality of the code. I'd be happy to discuss this further if you have any additional thoughts.

Reply:
done in 0446780e896c90777b485fee0efd82fef7f3ade2, please check if the change is what you had in mind